### PR TITLE
Add query action to UI 38

### PR DESCRIPTION
**Pull request by bkudler**
- Adds querying ability on front end
- fixes cucumber tests to work for home page
- all tests passing for feature
- adds ability to query start_date and end_date

### CODE REVIEW

File: features/steps/promotions_steps.py [hunk 1]
```diff
@@ -32,9 +38,79 @@ def step_impl(context):
...
-            "promotion_changes_price": row['promotion_changes_price'] in ['True', 'False']
+            "promotion_changes_price": row['promotion_changes_price'] in ['True', 'true', '1'],
+            "has_been_extended": row['promotion_changes_price'] in ['True', 'true', '1'],
+            "original_end_date": row['end_date'],
         }
         context.resp = requests.post(rest_endpoint, json=payload)
         assert(context.resp.status_code == HTTP_201_CREATED)
+
+
+@given(u'the server is started')
```
Comment (loixe) [inline features/steps/promotions_steps.py:51]: I'm not sure, but I saw these lines (51-116) in web_steps.py in Professor's lab.

File: service/routes.py [hunk 2]
```diff
@@ -108,10 +109,16 @@ def list_promotions():
     promotions = []
     message = request.args.get("message")
     name = request.args.get("name")
+    start_date = request.args.get("start_date")
+    end_date = request.args.get("end_date")
     if message:
         promotions = Promotion.find_by_message(message)
     elif name:
         promotions = Promotion.find_by_name(name)
+    elif start_date:
```
Comment (loixe) [inline service/routes.py:118]: maybe need to update test_routes.py as well

File: service/static/js/rest_api.js [hunk 3]
```diff
@@ -1,3 +1,106 @@
 $(function () {
+    function flash_message(message) {
+        $("#flash_message").empty();
+        $("#flash_message").append(message);
+    }
 
+    function update_form_data(res) {
+        $("#pet_id").val(res.id);
+        $("#pet_name").val(res.name);
+        $("#pet_category").val(res.category);
```
Comment (loixe) [inline service/static/js/rest_api.js:10]: here should be 'promotion'

File: features/steps/promotions_steps.py (hunk 1 above)
Comment (bkudler) [reply features/steps/promotions_steps.py:51]: I don't think it makes a difference.

**Pull request merged**